platform/source/posix/iot_threads_posix.c

```diff
@@ -137,23 +137,75 @@ static void * _threadRoutineWrapper( void * pArgument )
 /*-----------------------------------------------------------*/
 
 bool Iot_CreateDetachedThread( IotThreadRoutine_t threadRoutine,
-                               void * pArgument )
+                               void * pArgument,
+                               int32_t priority,
+                               size_t stackSize )
 {
     bool status = true;
     int posixErrno = 0;
+    _threadInfo_t * pThreadInfo = NULL;
     pthread_t newThread;
     pthread_attr_t threadAttributes;
```

Comment:
This definately needs to be a pointer and not an instance on the stack if you look at init and destroy. This will cause WEIRD behavior